File: ejercicios/src/main/java/co/edu/sena/ejercicios/ejercicio12/Factura.java
package co.edu.sena.ejercicios.ejercicio12;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Objects;

public class Factura {
    private int idFactura;
    private LocalDateTime fecha;
    private double totalFactura;
    private double iva;
    private Cliente cliente;
    private List<Item> items;

    public int getIdFactura() {
        return idFactura;
    }

    public void setIdFactura(int idFactura) {
        this.idFactura = idFactura;
    }

    public LocalDateTime getFecha() {
        return fecha;
    }

    public void setFecha(LocalDateTime fecha) {
        this.fecha = fecha;
    }

    public double getTotalFactura() {
        return totalFactura;
    }

    public void setTotalFactura(double totalFactura) {
        this.totalFactura = totalFactura;
    }

    public double getIva() {
        return iva;
    }

    public void setIva(double iva) {
        this.iva = iva;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public List<Item> getItems() {
        return items;
    }

    public void setItems(List<Item> items) {
        this.items = items;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Factura factura = (Factura) o;
        return idFactura == factura.idFactura;
    }

    @Override
    public int hashCode() {
        return Objects.hash(idFactura);
    }

    @Override
    public String toString() {
        return "Factura{" +
                "idFactura=" + idFactura +
                ", fecha=" + fecha +
                ", totalFactura=" + totalFactura +
                ", iva=" + iva +
                '}';
    }
}

File: typescript/src/ejercicio12/factura.ts
import {Cliente} from "./cliente";
import {Item} from "./item";

export class Factura {
    private _idFactura: number;
    private _fecha: Date;
    private _totalFactura: number;
    private _iva: number;
    private _cliente: Cliente;
    private _items: Array<Item>


    get idFactura(): number {
        return this._idFactura;
    }

    set idFactura(value: number) {
        this._idFactura = value;
    }

    get fecha(): Date {
        return this._fecha;
    }

    set fecha(value: Date) {
        this._fecha = value;
    }

    get totalFactura(): number {
        return this._totalFactura;
    }

    set totalFactura(value: number) {
        this._totalFactura = value;
    }

    get iva(): number {
        return this._iva;
    }

    set iva(value: number) {
        this._iva = value;
    }

    get cliente(): Cliente {
        return this._cliente;
    }

    set cliente(value: Cliente) {
        this._cliente = value;
    }

    get items(): Array<Item> {
        return this._items;
    }

    set items(value: Array<Item>) {
        this._items = value;
    }
}